lib/Doctrine/DBAL/Platforms/PostgreSqlPlatform.php

```diff
@@ -1236,4 +1236,12 @@ private function isNumericType(Type $type) : bool
     {
         return $type instanceof IntegerType || $type instanceof BigIntType;
     }
+
+    /**
+     * {@inheritDoc}
+     */
+    protected function quoteDefaultStringLiteral(string $str) : string
+    {
+        return parent::quoteStringLiteral($str);
```

Comment:
Because [PostgreSQL does not support backlash C-style escapes out of the box](https://www.postgresql.org/docs/9.6/static/sql-syntax-lexical.html#SQL-SYNTAX-STRINGS-ESCAPE), the `quoteStringLiteral()` had to be overridden a long time ago. See this quick example:

```
db=# SELECT 'foo\bar', 'foo\\bar', E'foo\\bar';
 ?column? | ?column? | ?column? 
----------+----------+----------
 foo\bar  | foo\\bar | foo\bar
(1 row)
```

This means we cannot use `PostgreSqlPlatform::quoteStringLiteral()` for our use-case. And this is the only reason why we created a new `quoteDefaultStringLiteral()`, to be able to override it to "cancel" the original override.

I wish it was not needed, but I don't say another way around...

Reply:
scratch that last comment of mine, I'll think a bit more about that...

Reply:
The fact that `quoteStringLiteral()` is not overridden may mean that it's not tested well enough. You could try testing other (non-DDL) cases and see if those tests fail as well.

Reply:
I don't understand what you mean by that. `AbstractPlatform::quoteStringLiteral` is indeed overridden by `PostgreSqlPlatform::quoteStringLiteral`. But we intentionally use the non-overridden method in our code. And that is covered by tests.

Reply:
@PowerKiKi I don't remember what I meant by that :confused:.